misc-scripts/xref_mapping/XrefParser/XenopusJamboreeParser.pm

```diff
@@ -52,35 +52,61 @@ sub run {
     return 1;    # 1 error
   }
 
+  my $input_file = Text::CSV->new({
+                                    sep_char           => "\t",
+				    empty_is_undef     => 1,
+				   }) or croak "Cannot use file $file: ".Text::CSV->error_diag ();
+
+  $input_file->column_names([qw(acc label desc stable_id)] );
   my $count = 0;
-  while ( $_ = $file_io->getline() ) {
-    chomp;
-    my ($acc, $label, $desc, $stable_id) = split /\t/;
-    # Remove some provenance information encoded in the description
-    $desc =~ s/\[.*\]//;
-    # Remove labels of type 5 of 14 from the description
-    $desc =~ s/ , [0-9]+ of [0-9]+//;
-
-    if($label eq "unnamed"){
-      $label = $acc;
+  while ( my $data = $input_file->getline_hr( $file_io ) ) {
+
+    my $desc;
+    if(defined $data->{'desc'}){
+      $desc = $self->parse_description($data->{'desc'});
     }
 
-    $self->add_to_direct_xrefs({ stable_id  => $stable_id,
+    if($data->{'label'} eq "unnamed"){
+      $data->{'label'} = $data->{'acc'};
+    }
+
+    $self->add_to_direct_xrefs({ stable_id  => $data->{'stable_id'},
 				 type       => 'gene',
-				 acc        => $acc,
-				 label      => $label,
+				 acc        => $data->{'acc'},
+				 label      => $data->{'label'},
 				 desc       => $desc,
                                  dbi        => $dbi,
 				 source_id  => $source_id,
 				 species_id => $species_id });
     $count++;
   }
 
+  $input_file->eof or croak "Error parsing file $file: " . $input_file->error_diag();
   $file_io->close();
 
   print $count . " XenopusJamboreeParser xrefs succesfully parsed\n" if($verbose);
 
   return 0;
 }
 
+
+=begin comment
+Regex handles lines in the following desc formats
+
+XB-GENE-940410	unnamed	Putative ortholog of g2/mitotic-specific cyclin B3, 3 of 14	ENSXETG00000007206
+XB-GENE-956173	hba4	alpha-T4 globin, Putative ortholog of hemoglobin alpha chain. [Source:Uniprot/SWISSPROT;Acc:P01922], 2 of 3	ENSXETG00000001141
+
+=end comment
+=cut
+sub parse_description{
+  my ($self, $desc) = @_;
+
+  # Remove some provenance information encoded in the description
+  $desc =~ s/\[.*\]//;
+  # Remove labels of type 5 of 14 from the description
+  $desc =~ s/ , [0-9]+ of [0-9]+//;
```

Comment:
confirm this regex is doing what you want...
I find that space before the comma disturbing...